tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSSubscriptionConsumer.java

```diff
@@ -0,0 +1,216 @@
+                sender.declareQueue(QueueSpecification
+                    .queue(queue)
+                    .durable(DURABLE)
+                    .arguments(queueArgumentSupplier()
+                        .deadLetter(deadLetter)
+                        .singleActiveConsumer()
+                        .consumerTimeout(Duration.ofMinutes(10L).toMillis())
+                        .build())),
+                sender.bind(BindingSpecification.binding()
+                    .exchange(exchange)
+                    .queue(queue)
+                    .routingKey(saasSubscriptionRabbitMQConfiguration.routingKey())))
+            .then()
+            .block();
+    }
+
+    private Mono<AMQP.Exchange.DeclareOk> declareExchange(String exchange) {
+        return sender.declareExchange(ExchangeSpecification.exchange(exchange)
+                .durable(DURABLE).type(BuiltinExchangeType.TOPIC.getType()))
+            .onErrorResume(error -> error instanceof ShutdownSignalException && error.getMessage().contains("reply-code=406, reply-text=PRECONDITION_FAILED"),
+                error -> {
+                    LOGGER.warn("Exchange `{}` already exists but with different configuration. Ignoring this error. \nError message: {}", exchange, error.getMessage());
+                    return Mono.empty();
+                });
+    }
+
+    private QueueArguments.Builder queueArgumentSupplier() {
+        if (!twpCommonRabbitMQConfiguration.quorumQueuesBypass()) {
+            return rabbitMQConfiguration.workQueueArgumentsBuilder();
+        }
+        return QueueArguments.builder();
+    }
+
+    public void startConsumer() {
+        consumeSubscriptionDisposable = consumeSubscriptionQueue();
+    }
+
+    public void restartConsumer() {
+        Disposable previousConsumer = consumeSubscriptionDisposable;
+        consumeSubscriptionDisposable = consumeSubscriptionQueue();
+        Optional.ofNullable(previousConsumer)
+            .ifPresent(Disposable::dispose);
+    }
+
+    private Disposable consumeSubscriptionQueue() {
+        return delivery(SAAS_SUBSCRIPTION_QUEUE)
+            .concatMap(delivery -> consumeSubscriptionUpdate(delivery, delivery.getBody()))
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe();
+    }
+
+    public Flux<AcknowledgableDelivery> delivery(String queue) {
+        return Flux.using(receiverProvider::createReceiver,
+            receiver -> receiver.consumeManualAck(queue, new ConsumeOptions()
+                .qos(DEFAULT_CONCURRENCY)),
+            Receiver::close);
+    }
+
+    private Mono<Void> consumeSubscriptionUpdate(AcknowledgableDelivery ackDelivery, byte[] messagePayload) {
+        return Mono.fromCallable(() -> SaaSSubscriptionMessage.Deserializer.parseAMQPMessage(messagePayload))
+            .flatMap(this::handleSubscriptionMessage)
+            .doOnSuccess(result -> {
+                LOGGER.debug("Consumed SaaS subscription message successfully");
```

Comment:
```suggestion
                LOGGER.info("Consumed SaaS subscription message successfully for user {}", user);
```

Would be quite helpful IMO